Cannot build the website: CalendarLib/Syndic issue?

I'm trying to build the website.

(Do we really require ocaml 4.01.0, as enforced by the `opam` file? Is there an issue discussing what would be required to more recent OCaml versions?)

After a bit of debugging of the opam-repository dependencies, I got to this error

```
make staging SET_STAGING='-set staging'
make[1]: Entering directory '/home/gasche/Prog/ocaml/ocaml.org'
cd script && \
ocamlfind ocamlopt -annot -g -package netstring,nettls-gnutls,netclient,syndic \
  -linkpkg utils.cmx http.ml rss2html.ml -o ../"script/rss2html"
File "rss2html.ml", line 126, characters 33-57:
Error: Unbound module CalendarLib
Makefile.common:58: recipe for target 'script/rss2html' failed
```

If I apply the following diff

```diff
diff --git a/Makefile.common b/Makefile.common
index b333752..a10cc63 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -56,7 +56,7 @@ script/breadcrumb: script/breadcrumb.ml $(UTILS_DEP)
 
 script/rss2html: script/rss2html.ml script/http.ml $(UTILS_DEP)
        cd script && \
-       $(OCAMLOPT) -package netstring,nettls-gnutls,netclient,syndic \
+       $(OCAMLOPT) -package netstring,nettls-gnutls,netclient,syndic,calendar \
          -linkpkg utils.cmx http.ml rss2html.ml -o ../"$@"
 
 script/ocamltohtml:script/lexer.ml script/ocamltohtml.ml
diff --git a/opam b/opam
index a496d1a..829b3a8 100644
--- a/opam
+++ b/opam
@@ -19,6 +19,7 @@ depends: [
   "ocamlnet" {>= "4.0.1"}
   "conf-gnutls"
   "mpp" {>= "0.1.2"}
+  "calendar"
   "ocamlrss"
   "uri" {>= "1.3.11"}
   "syndic" {>= "1.4"}
```

then I get `calendar 2.03.2` and the following build error:

```
cd script && \
ocamlfind ocamlopt -annot -g -package netstring,nettls-gnutls,netclient,syndic,calendar \
  -linkpkg utils.cmx http.ml rss2html.ml -o ../"script/rss2html"
File "rss2html.ml", line 126, characters 32-60:
Error: This expression has type CalendarLib.Calendar.t
       but an expression was expected of type Syndic.Atom.updated = Ptime.t
```

--- a/.gitignore
+++ b/.gitignore
@@ -9,6 +9,7 @@ configure
 *.cm[ioax]
 *.cmxa
 *.annot
+.*.swp
 doclib.docdir
 ocaml.org/
 ocamltohtml

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,8 @@
+FROM ocaml/opam
+RUN git -C /home/opam/opam-repository pull && opam update -u -y
+COPY . /home/opam/src
+RUN sudo chown -R opam /home/opam/src
+RUN opam pin add -n ocaml.org /home/opam/src
+RUN opam depext -u ocaml.org
+RUN opam install -j 4 -y -v --deps-only ocaml.org
+RUN opam install -y -v ocaml.org

--- a/Makefile.common
+++ b/Makefile.common
@@ -56,7 +56,7 @@ script/breadcrumb: script/breadcrumb.ml $(UTILS_DEP)
 
 script/rss2html: script/rss2html.ml script/http.ml $(UTILS_DEP)
 	cd script && \
-	$(OCAMLOPT) -package netstring,nettls-gnutls,netclient,syndic \
+	$(OCAMLOPT) -package netstring,nettls-gnutls,netclient,syndic,calendar \
 	  -linkpkg utils.cmx http.ml rss2html.ml -o ../"$@"
 
 script/ocamltohtml:script/lexer.ml script/ocamltohtml.ml

--- a/opam
+++ b/opam
@@ -18,10 +18,11 @@ depends: [
   "omd" {>= "1.2.1"}
   "ocamlnet" {>= "4.0.1"}
   "conf-gnutls"
+  "calendar"
   "mpp" {>= "0.1.2"}
   "ocamlrss"
   "uri" {>= "1.3.11"}
-  "syndic" {>= "1.4"}
+  "syndic" {>= "1.4" & <"1.5"}
   "camlp4"
 ]
 depopts: [
@@ -29,5 +30,5 @@ depopts: [
   "lablgtk2"
 ]
 available: [
-  ocaml-version = "4.01.0"
+  ocaml-version >= "4.01.0"
 ]

--- a/site/learn/tutorials/camlp4_3.10/pa_foreach3.ml
+++ b/site/learn/tutorials/camlp4_3.10/pa_foreach3.ml
@@ -12,7 +12,7 @@ let rec mkfun _loc patts e =
 
 let lident_of_patt p =
   match p with
-  | <:patt< $lid:i$ >> -> i
+  | <:patt< $i$ >> -> i
   | _ -> invalid_arg "lident_of_patt"
 
 let () =

